src/app.py

```diff
@@ -35,78 +39,68 @@ def license_page():
     # Check for badge query params and load if it exists.
     license = request.args.get('license')
-    html = ""
-    if license:
-        html = dataset.license_lookup(license)
+    html = dataset.license_lookup(license)
 
     # Return the basic lookup form if not
     return render_template("index.html", **LICENSE_CONTEXT, entity_html=html)
 
 
-@app.route("/license-lookup/<license>")
-def license_lookup(license):
-    html = dataset.license_lookup(license)
-    return render_template("index.html", **LICENSE_CONTEXT, entity_html=html)
-
-
-################################################################################
-# Badges
-################################################################################
-BADGE_CONTEXT = {
-    "title": "Seattle Officer Badge Lookup",
-    "entity_name_long": "badge number",
-    "entity_name_short": "Badge #",
-    "data_source": "https://data.seattle.gov/City-Business/City-of-Seattle-Wage-Data/2khk-5ukd",
-    "lookup_url": "badge",
-    "query_param": "badge"
-}
-
-
-@app.route("/badge")
-def badge_page():
-    """Badge lookup form render"""
-    # Check for badge query params and load if it exists.
-    badge = request.args.get('badge')
-    html = ""
-    if badge:
-        html = dataset.badge_lookup(badge)
-
-    return render_template("index.html", **BADGE_CONTEXT, entity_html=html)
-
-
-@app.route("/badge-lookup/<badge>")
-def badge_lookup(badge):
-    html = dataset.badge_lookup(badge)
-    return render_template("index.html", **BADGE_CONTEXT, entity_html=html)
-
-
 ################################################################################
 # Names
 ################################################################################
 NAME_CONTEXT = {
     "title": "Seattle Officer Name Lookup",
-    "entity_name_long": "last name",
-    "entity_name_short": "Lastname",
+    "entities": [
+        {
+            "entity_name_long": "last name",
+            "entity_name_short": "Last Name",
+            "query_param": "last"
+        },
+        {
+            "entity_name_long": "first name",
+            "entity_name_short": "First Name",
+            "query_param": "first"
+        },
+        {
+            "entity_name_long": "badge number",
+            "entity_name_short": "Badge Number",
+            "query_param": "badge"
+        }
+    ],
     "data_source": "https://data.seattle.gov/City-Business/City-of-Seattle-Wage-Data/2khk-5ukd",
-    "lookup_url": "name",
-    "query_param": "last"
+    "lookup_url": "name"
 }
 
 
 @app.route("/name")
 def name_page():
     """Officer name lookup form render"""
-    last_name = request.args.get('last')
-    html = ""
-    if last_name:
-        html = dataset.name_lookup(last_name)
+    last_name = None if request.args.get('last') == "" else request.args.get('last')
+    first_name = None if request.args.get('first') == "" else request.args.get('first')
+    badge = None if request.args.get('badge') == "" else request.args.get('badge')
```

Comment:
For each of these, I think we could use the following logic to simplify this:
```suggestion
    last_name = request.args.get('last') or None
    first_name = request.args.get('first') or None
    badge = request.args.get('badge') or None
```

Ex:
```
[ins] In [6]: def assign(x):
         ...:     print(x or None)
         ...: 

[ins] In [7]: assign("foo")
foo

[ins] In [8]: assign("")
None
```

Do these need to be `None`? Is there a reason they can't be an empty string?

Reply:
Brief explanation for why they need to be `None`:

```
records = sql_curs.execute(
                "SELECT * FROM officers WHERE Last = ? OR First = ? OR Serial = ?",
                (last_name, first_name, badge),
            ).fetchall()
```

You can see here that I'm using the sqlite library's cursor object, and doing a safe injection of the values to filter the query on from the `(last_name, first_name, badge)` tuple into the `?`s in order. The way the sqlite library protects against SQL injection attacks here is it wraps any strings injected in `"`s so that you can't escape out, so, when you pass any string object, even if it's empty, it gets wrapped in quotes, so you end up with:

`SELECT * FROM officers WHERE Last = "" OR First = "" OR Serial = ""`

Which will match empty cells instead of skipping the column. When the variables are just straight up `None` (as opposed to `str(None)`), you get this I think:

`SELECT * FROM officers WHERE Last =  OR First =  OR Serial = `

Which will just skip the column searches since no value is passed.

This explanation is provided for completeness we already discussed elsewhere.